### PR TITLE
MIME identification using python

### LINKED ISSUE #47: Document translation: detect mime type within python?
http://stackoverflow.com/a/19682945/69663 gives php code to do it for docx; it'd be nice not having to shell out to mimetype/xdg-mime/file (which have had [security](https://www.freebsd.org/security/advisories/FreeBSD-SA-14:16.file.asc) [issues](https://www.freebsd.org/security/advisories/FreeBSD-SA-14:28.file.asc)). We would only need to check for the 3 or 4 types we accept, and disregard the rest.

### PR DESCRIPTION
**Pull request by XDRAGON2002**
Fixes #47 
Earlier subprocesses mimetype/xdg-mime/file were being used to detect MIME type which could lead to security issues so now then have been switched with python-magic a library used to detect MIME type within python.

### CODE REVIEW

**Review by sushain97 — CHANGES_REQUESTED** [review]
Thanks for the PR! Please update the `Pipfile`, `setup.py` and `README.md` to reflect this new dependency.

**Comment by XDRAGON2002** [comment]
@sushain97 Updated Pipfile, setup.py and README.md and added python-magic as dependency.

**Review by sushain97 — APPROVED** [review]
Seems reasonable to me! I'll leave it to @unhammer since he opened the initial ticket.

**Comment by XDRAGON2002** [comment]
Great, Thanks for the review @sushain97 !

File: apertium_apy/handlers/translate_doc.py [hunk 1]
```diff
@@ -56,20 +49,11 @@ async def translate_doc(file_to_translate, fmt, mode_file, unknown_marks=False):
...
-                if shutil.which(command):
-                    cls.mime_type_command = command
-                    break
-
-        if not cls.mime_type_command:
-            return None
-        mime_command = MIMETYPE_COMMANDS[cls.mime_type_command]
-        mime_type = mime_command(f).split(';')[0]
+        mime = magic.Magic(mime = True)
+        mime_type = mime.from_file(f).split(';')[0]
```
Comment (unhammer) [inline apertium_apy/handlers/translate_doc.py:56]: What if this list is empty?

Comment (XDRAGON2002) [reply apertium_apy/handlers/translate_doc.py:56]: We could check if the list is empty and return None in that case?

Comment (XDRAGON2002) [reply apertium_apy/handlers/translate_doc.py:56]: We could check if the list is empty and return None in that case?

**Comment by unhammer** [comment]
Using libmagic instead of shelling out to file is nice, but security-wise it's not a huge improvement. As I wrote in #47, we only need a few types (current list has 10). `python-magic` is a thin layer of the libmagic C library with a huge range of file detection capabilities and security bug potential.

> We're not sure whether any distribution has packages that rely on server-side use of libmagic, or whether it's common to have long-running processes that use libmagic with untrusted input
https://seclists.org/oss-sec/2014/q1/504  hey, they're talking about us :)

I was originally thinking it might be easy to just manually do the checks (our hfst bins start with `HFST`, how hard can it be to parse just 10 types), but e.g. rtf's can start with `{rtf` or `{pard` and who knows what else and there are probably even worse annoyances with docx etc. So maybe that's not the way to go.

A better alternative I think would be to just restrict the magic database. It seems the `python-magic` lib supports loading a user-specified magic database. You can compile a `.mgc` with `file -C -m ./dir-with-only-our-few-types/` (and `apt-get source libmagic-dev` gives you the relevant pattern files in `magic/Magdir/msooxml` etc.), test that it doesn't detect other things with `file -m .mgc somefile.pdf`.

**Pull request closed without merging**